### DISCUSSION TITLE
軟體生命週期

### DISCUSSION THREAD

**Discussion by seastar0687** [1101-GS4719-C/Python-Community-GS4719-C#3698] [General]
在其他課堂看到有關軟體生命週期的知識，覺得對大家的期末專題有用處，故來分享
軟體生命週期分成四階段: 分析，設計，開發，測試
分析階段 (Analysis Phase): 確定軟體需要實現什麼功能，不關注如何實現
設計階段 (Design Phase): 定義系統如何實現分析階段中確定的功能
實作階段 (Implementation Phase): 實現設計中的模組或類別
測試階段 (Testing Phase): 找出軟體中的錯誤

以上是軟體生命週期，這住要用在開發各種軟體上，如果大家要開發軟體可以參考這個流程來實踐
我個人在做期末專題時也是這樣實踐的，因為根據這個方法實踐會是更加標準化，同時也是增強程式能力的好訓練。

**Comment by kai1660** [comment]
非常有趣的介紹:軟體生命週期，大家要開發軟體可以參考這個流程來實踐，依據上述的四個階段，我們可以更有條理地分析我們所製作的軟體，我希望我的期末專題可以運用這一套方實踐，這一個方法推薦給大家。
